peer.go

```diff
@@ -1171,13 +1171,46 @@ out:
 
 			// Make sure this channel is not already active.
 			p.activeChanMtx.Lock()
-			if _, ok := p.activeChannels[chanID]; ok {
+			if currentChan, ok := p.activeChannels[chanID]; ok {
 				peerLog.Infof("Already have ChannelPoint(%v), "+
 					"ignoring.", chanPoint)
+
```

Comment:
Should closing of the `newChanReq.done` be moved to the end? In case the sender is synchronising on this and expects the `breachArbiter` to have the updated contract and revocation to be set up when it closes.

Reply:
Might have to close at each return/continue as well?

Reply:
Atm, the caller (the funding manager) has already sent a channel to the breach arbiter. We then override that by sending out this _newer_ channel who's pointer to the next revocation point will be updated in memory.